test/functional_execution_context/tests/execution_context.ts

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { Ecs, KibanaExecutionContext } from 'kibana/server';
+
+import Fs from 'fs/promises';
+import Path from 'path';
+import { isEqual } from 'lodash';
+import type { FtrProviderContext } from '../ftr_provider_context';
+
+const logFilePath = Path.resolve(__dirname, '../kibana.log');
+
+// to avoid splitting log record containing \n symbol
+const endOfLine = /}\s*\n/;
```

Comment:
```suggestion
const endOfLine = /(?<=})\s*\n/;
```

Would allow you to split only on newlines that follow `}` without also stripping the `}`.